OneCommunityGlobal/HighestGoodNetworkApp
# yihan fix weekly summary submission date

# Description
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94303659/0de0c3df-94a0-494b-af2f-0b3d8f098086)

## Related PRS (if any):
This frontend PR is related to the[ #359 ](https://github.com/OneCommunityGlobal/HGNRest/pull/359)backend PR.

## Main changes explained:
Added uploadDate in weeklySummaries object. The default uploadDate(when there's no uploaded summary) is the end of this week. After submitting the weekly summary, the uploadDate turns to the submission date.
On weekly summary report page, it will show the uploadDate instead of the dueDate.

## How to test:
check into current branch on both frontend and backend
do `npm run start:local` to run this PR locally
log as owner/admin user
first go to dashboard and upload a summary
go to Reports → Weekly Summaries Report
check the date next to Weekly Summary should be the date the person uploaded the summary. If there's no uploaded summary, the date should be the due date.

## Screenshots or videos of changes:
This Week:
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94303659/4d27a94d-807e-48a2-94d0-9d5a210b2655)

Last Week:
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94303659/19d926fb-5ea9-452a-a0a7-1b9466f2ff7e)

Week Before Last:
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94303659/0047b7ab-5e34-45d1-b103-ea485c0a5aa0)

Three Weeks Ago:
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94303659/850d8833-3233-4027-affc-19ab45af9c97)
